jsonParser.hs

```diff
@@ -3,15 +3,15 @@
 data JSON = N Double | S String | B Bool | Null | L [JSON] | O [(String, JSON)]
     deriving (Show, Eq)
 
-data Parser a = (a, String)
+type Parser a = (a, String)
```

Comment:
I'm not sure why, but ghci was complaining about overriding a builtin operator `,`.  Or something like that.  Try changing it back and run it.  You'll get the error.